src/test/java/org/casbin/jcasbin/main/BuiltInFunctionsUnitTest.java

```diff
@@ -61,4 +63,53 @@ public void testGlobMatchFunc() {
         testGlobMatch("/prefix/subprefix/foobar", "*/foo*", false);
         testGlobMatch("/prefix/subprefix/foobar", "*/foo/*", false);
     }
+
+    @Test
+    public void testKeyGetFunc() {
+        testKeyGet("/foo","/foo","");
+        testKeyGet("/foo","/foo*","");
+        testKeyGet("/foo", "/foo/*", "");
+        testKeyGet("/foo/bar", "/foo", "");
+        testKeyGet("/foo/bar", "/foo*", "/bar");
+        testKeyGet("/foo/bar", "/foo/*", "bar");
+        testKeyGet("/foobar", "/foo", "");
+        testKeyGet("/foobar", "/foo*", "bar");
+        testKeyGet("/foobar", "/foo/*", "");
+    }
+
+    @Test
+    public void TestKeyGet2() {
+        testKeyGet2("/foo", "/foo", "id","");
+        testKeyGet2("/foo", "/foo*", "id","");
+        testKeyGet2("/foo", "/foo/*", "id","");
+        testKeyGet2("/foo/bar", "/foo", "id","");
+        // different with KeyMatch.
+        testKeyGet2("/foo/bar", "/foo*", "id","");
+        testKeyGet2("/foo/bar", "/foo/*", "id","");
+        testKeyGet2("/foobar", "/foo", "id","");
+        // different with KeyMatch.
+        testKeyGet2("/foobar", "/foo*", "id","");
+        testKeyGet2("/foobar", "/foo/*", "id","");
+
+        testKeyGet2("/", "/:resource", "resource","");
+        testKeyGet2("/resource1", "/:resource", "resource","resource1");
+        testKeyGet2("/myid", "/:id/using/:resId", "id","");
+        testKeyGet2("/myid/using/myresid", "/:id/using/:resId", "id","myid");
```

Comment:
code format.

Reply:
Sorry, I don't know what is the problem of this code format.

Reply:
A space after comma (,), many lines have such problem.